src/vcpkg/vcpkgpaths.cpp

```diff
@@ -834,6 +834,13 @@ namespace vcpkg
             .map([](Path&& dot_git_parent) { return std::move(dot_git_parent) / ".git"; });
     }
 
+    bool VcpkgPaths::is_overlay_port(StringView port_name) const
+    {
+        return std::find_if(overlay_ports.begin(), overlay_ports.end(), [&port_name](StringView port_dir) {
```

Comment:
This isn't right either because direct overlay-ports don't care about the directory name. Can you just revert the behavior for this one to how it was before?